### DISCUSSION TITLE
Require package to fail if PHP > 7.4

### DISCUSSION THREAD

**Discussion by bilogic** [composer/composer#11321] [Q&A / Support]
I have a legacy package can only run up til PHP 7.4, so I would like its `composer require` to fail if the OS is running any later PHP version.

I tried 
```
    "require": {
        "php": "^7",
...
```

But the package still gets downloaded without error. Any idea how I can achieve this? Thank you.

**Comment by Seldaek** [comment]
Does it revert to an old version? Can you post repro instructions or at least the package name?

**Reply by bilogic** [reply]
They are all private packages. I do understand why repro will help, let me see what I can do.

For now, perhaps I can rephrase the problem/question:

1. The package has `"php": "^7"`, which I understand to mean `>=7 - <8`, must be within 7 (cannot be 8 and above)
2. Composer is running on PHP 8.1
3. What should be the behavior of composer? Install the package? Or throw an error?

**Reply by bilogic** [reply]
I recreated a minimal set of repositories and it behaved as I expected, so it's probably an issue in my private packages.

Thank you.

```
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - dev1/a dev-master requires php ^7.0 -> your php version (8.1.14) does not satisfy that requirement.
    - dev1/b dev-master requires dev1/a * -> satisfiable by dev1/a[dev-master].
    - Root composer.json requires dev1/b * -> satisfiable by dev1/b[dev-master].
```

**Reply by bilogic** [reply]
I feel this discussion won't help anyone and can be deleted as "garbage collection". Thank you.

**Reply by Seldaek** [reply]
Maybe you have a config.platform set to php 7? That'd let it resolve but it should still fail a composer check-platform-reqs

**Reply by bilogic** [reply]
@Seldaek yes I did, that was the cause of my problem. Thank you.
